### ISSUE TITLE
No way to discern unhandled exception return value versus number of failed tests

### ISSUE THREAD

**Issue by MattGal** [xunit/xunit#659]
Currently, both in https://github.com/xunit/xunit/blob/master/src/xunit.console/Program.cs and the dotnet .NET Core version of the XUnit console runner, all return values are positive, with the return value from the exe being either 1 if two different types of exception are seen, or the number of test failures (which can be anywhere from 0.. some large integer), or another number if execution fails in a way that isn't covered by the two catch blocks.

In the case of an indirect dependency failing to resolve at runtime, XUnit will return values like 255 to the caller.  It would be really nice to get a simple rule in place... such as "all low-level failures give negative numbers" or otherwise that makes it possible to use the return value of XUnit in a meaningful way.  The way it exists today, it seems I must always check for the results xml in addition to any check made to be sure the expected thing happened.

**Comment by bradwilson** [comment]
Negative return values don't work well with CMD and error level testing. We can standardize on error codes, though.

**Comment by MattGal** [comment]
I figured that was the reason.   It's just challenging to automate checking for the outcome without first checking for the results output xml existing and then seeing if the exit code 1 (or other, for unhandled exception types) means 1 failure or exception by checking the output.   That's how we've worked around it, so it's not blocking... but I'd love to see this improved in the code, or someone to provide a better pattern for dealing with this than the one we've come up with.
